Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -135,5 +142,164 @@ lemma map [TopologicalSpace 𝓨] [MeasurableSpace 𝓨] [OpensMeasurableSpace 
+        refine (μ (sub b)).apply_mono
+            <| Set.biUnion_mono (fun i (hi : i ≤ n) ↦ hi.trans ?_) fun _ _ ↦ le_rfl
+        exact le_trans (Nat.le_add_right n 1) (le_trans hypo (StrictMono.le_apply hsubmono))
+      · use 0; simp
+      · use 1
+        simpa [ge_iff_le, eventually_map, eventually_atTop, ge_iff_le, forall_exists_index] using
+            fun _ d hyp ↦ (hyp d (by simp)).trans
+            <| ProbabilityMeasure.apply_le_one (μ (sub d)) (⋃ i ≤ sub d, U i)
+    _ ≤ 1 - ε := by
+      apply Filter.liminf_le_of_le
+      · use 0; simp
+      simp only [eventually_atTop, ge_iff_le, forall_exists_index]
+      intro b c h
+      apply le_trans (h c le_rfl)
+      refine (ofReal_le_ofReal_iff (by rw [sub_nonneg]; exact hεbound)).mp ?_
+      rw [ofReal_coe_nnreal]
+      apply le_trans (hcontradiction (sub c))
+      norm_cast
+  have accumulation : Tendsto (fun n ↦ μlim (⋃ i ≤ n, U i)) atTop (𝓝 (μlim (⋃ i, U i))) := by
+    simp_rw [←Set.accumulate_def, ProbabilityMeasure.tendsto_measure_iUnion_accumulate]
+  rw [Cov, coeFn_univ, ← NNReal.tendsto_coe] at accumulation
+  have exceeds_bound : ∀ᶠ n in atTop, (1 - ε / 2 : ℝ) ≤ μlim (⋃ i ≤ n, U i) :=
+      Tendsto.eventually_const_le (v := 1) (by simp; positivity) (accumulation)
+  suffices ∀ᶠ n : ℕ in atTop, False by exact this.exists.choose_spec
+  filter_upwards [exceeds_bound] with n hn
+  linarith [hn.trans <| Measurebound n]
+
+variable [CompleteSpace 𝓧]
+
+theorem isTightMeasureSet_of_isCompact_closure (hcomp : IsCompact (closure S)) :
+    IsTightMeasureSet {((μ : ProbabilityMeasure 𝓧) : Measure 𝓧) | μ ∈ S} := by
+  rw [IsTightMeasureSet_iff_exists_isCompact_measure_compl_le]
+  rcases isEmpty_or_nonempty 𝓧 with hempty | hnonempty
+  · intro ε εpos
+    use ∅
+    constructor
+    · exact isCompact_empty
+    rw [← univ_eq_empty_iff] at hempty
+    simp [mem_setOf_eq, compl_empty, hempty, measure_empty, zero_le, implies_true]
+  obtain ⟨D, hD⟩ := exists_dense_seq 𝓧
+  obtain ⟨u, hu_anti, hu_pos, hu⟩ : ∃ u, StrictAnti u ∧ (∀ n, 0 < u n) ∧ Tendsto u atTop (𝓝 0) :=
+    exists_seq_strictAnti_tendsto (0 : ℝ)
+  have hcov (m : ℕ) : ⋃ i, ball (D i) (u m) = univ := by
+    rw [denseRange_iff] at hD
+    ext p
+    exact ⟨fun a ↦ trivial, fun _ ↦ mem_iUnion.mpr <| hD p (u m) (hu_pos m)⟩
+  intro ε εpos
+  by_cases hεbound : 1 < ε
+  · use ∅
+    constructor
+    · exact isCompact_empty
+    intro μ hμ
+    simp only [mem_setOf_eq] at hμ
+    obtain ⟨μ', hμ', rfl⟩ := hμ
+    rw [compl_empty,measure_univ]
+    exact le_of_lt hεbound
+  have byclaim (m : ℕ) : ∃ k, ∀ μ ∈ S, 1 - (ε * 2 ^ (- m : ℤ) : ℝ≥0∞) <
+      μ (⋃ i ≤ k, ball (D i) (u m)) := by
+    refine (MeasOpenCoverTendstoMeasUniv (S := S) (U := fun i ↦ ball (D i) (u m))
+      (ε := (ε * 2 ^ (-m : ℤ))) (hε := ?_) (fun i ↦ isOpen_ball) hcomp) ?_ (hcov m)
+    · simpa using ⟨εpos, (ENNReal.zpow_pos (by simp) (by simp) (-↑m))⟩
+    · exact Left.mul_le_one (le_of_not_gt hεbound) <| zpow_le_one_of_nonpos (by linarith) (by simp)
```

Comment:
Reorder the arguments, adapt to the new type of `hεbound` from the change above.
```suggestion
    refine MeasOpenCoverTendstoMeasUniv (S := S) (fun i ↦ ball (D i) (u m))
      (fun _ ↦ isOpen_ball) (hcov m) hcomp (ε * 2 ^ (-m : ℤ)) ?_ ?_
    · simpa using ⟨εpos, (ENNReal.zpow_pos (by simp) (by simp) (-↑m))⟩
    · exact Left.mul_le_one hεbound <| zpow_le_one_of_nonpos (by linarith) (by simp)
```